### PR TITLE
HCAL Digi Filtering Module

### PR DESCRIPTION
**Pull request by Sam-Harper**
Dear All,

The E/gamma HLT needs to run method 2 on the HCAL Digis, however to run on all of them is too slow. So we have made a module which filters the HCAL digis around our reco::RecoEcalCandidates. We have tried to make it generic. 

One thing I wasnt so sure about was where to stick the helper classes:
EtaPhiRegion
EtaPhiRegionData (and its base class)
As this is producer which wont be included by anything, its not doing harm in same header but I dont like it. Any suggestions on where to stick them (or just ignore it and leave it be).

We will need this before we can deploy our new H/E variable although its not an urgent bug fix, more an desired improvement

Cheers,
Sam

@gpasztor FYI

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Sam-Harper** [comment]
If anybody is wondering, it looks like the checks failed due an xrootd failure. This module cant actually fail checks as it adds a brand new module that no code uses :)

**Comment by Sam-Harper** [comment]
Can we redo the tests? I think it was just an xrootd fail

**Comment by perrotta** [comment]
please test

File: RecoEgamma/EgammaHLTProducers/interface/HLTCaloObjInRegionsProducer.h [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    return new EtaPhiRegionData<l1t::EGammaBxCollection>(para,consumesColl);
+  }else if(type=="L1Jet"){
+    return new EtaPhiRegionData<l1t::JetBxCollection>(para,consumesColl);
+  }else if(type=="L1Muon"){
+    return new EtaPhiRegionData<l1t::MuonBxCollection>(para,consumesColl);
+  }else if(type=="L1Tau"){
+    return new EtaPhiRegionData<l1t::TauBxCollection>(para,consumesColl);
+  }else if(type=="RecoEcalCandidate"){
+    return new EtaPhiRegionData<reco::RecoEcalCandidateCollection>(para,consumesColl);
+  }else if(type=="RecoChargedCandiate"){
```
Comment (perrotta) [inline RecoEgamma/EgammaHLTProducers/interface/HLTCaloObjInRegionsProducer.h]: "RecoChargedCandidate"?

**Comment by Sam-Harper** [comment]
thanks for the spot, updated.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**